### PR TITLE
watson: add module

### PR DESCRIPTION
**Pull request by polykernel**
Watson is a CLI for tracking your time.

Two unit tests were added to validate the module behavior for
an empty configuration and the example configuration.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/watson.nix [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    "Library/Application Support"
+  else
+    config.xdg.configHome;
+
+in {
+  options = {
+    programs.watson = {
+      enable = mkEnableOption ''
+        watson, a wonderful CLI to track your time!
+      '';
```
Comment (sumnerevans) [inline modules/programs/watson.nix:21]:
```suggestion
      enable = mkEnableOption "watson, a CLI to track your time'';
```

File: modules/programs/watson.nix [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        '';
+      };
+
+      enableFishIntegration = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether to enable Watson's Fish integration.
+        '';
+      };
```
Comment (sumnerevans) [inline modules/programs/watson.nix:52]: these should use `mkEnableOption`:

```suggestion
      enableBashIntegration = mkEnableOption "watson's bash integration" // {
        default = true;
      };

      enableZshIntegration = mkEnableOption "watson's zsh integration" // {
        default = true;
      };

      enableFishIntegration = mkEnableOption "watson's fish integration" // {
        default = true;
      };
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]

File: modules/programs/watson.nix [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+with lib;
+
+let
+
+  cfg = config.programs.watson;
+
+  iniFormat = pkgs.formats.ini { };
+
+  configDir = if pkgs.stdenv.isDarwin then
```
Comment (berbiche) [inline modules/programs/watson.nix:11]:
```suggestion
  configDir = if pkgs.stdenv.hostPlatform.isDarwin then
```

File: modules/programs/watson.nix [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    "Library/Application Support"
+  else
+    config.xdg.configHome;
+
+in {
+  options = {
+    programs.watson = {
+      enable = mkEnableOption "watson, a wonderful CLI to track your time!";
```
Comment (berbiche) [inline modules/programs/watson.nix:19]:
```suggestion
      enable = mkEnableOption "watson, a wonderful CLI to track your time";
```
A period is automatically appended.

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks.

Found additional nitpicks.

File: modules/programs/watson.nix (hunk 3 above)
Comment (polykernel) [reply modules/programs/watson.nix]: thanks! I have also updated the tests.

File: modules/programs/watson.nix (hunk 4 above)
Comment (polykernel) [reply modules/programs/watson.nix]: thanks!

**Review by berbiche — APPROVED** [review]
LGTM!

ping @sumnerevans

**Pull request merged**